### PR TITLE
fenced in revamp (gold)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
<sup><sub>my hands will be their hands until im free</sub></sup>

---
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: solutions/usaco-623.mdx [hunk 1]
```diff
@@ -2,115 +2,140 @@
...
-**Time Complexity:** $\mathcal{O}((N\text{max}(y_i))(\text{log}(\text{max}(y_i))+\text{log}(N)))$
+**Time Complexity:** $\mathcal{O}((N \cdot  \max \{ y_i = 1 .. n \}
+)(\text{log}(\max \{ y_i = 1 .. n \})+\text{log}(N)))$
 
 <LanguageSection>
 <CPPSection>
 
 ```cpp
-#include <bits/stdc++.h>
+#include "bits/stdc++.h"
```
Comment (envyaims) [inline solutions/usaco-623.mdx:25]: ?

File: solutions/usaco-623.mdx [hunk 2]
```diff
@@ -2,115 +2,140 @@
...
+	 * -------------
+	 * |  1  |  2  |
+	 * |-----------|
+	 * |  3  |  4  |
+	 * -------------
+	 */
+	vector<Edge> edges;
+
+	// vertical fence edges
+	int sct = 0, e = 1;
```
Comment (envyaims) [inline solutions/usaco-623.mdx:97]: wat does these stand for

File: solutions/usaco-623.mdx [hunk 3]
```diff
@@ -2,115 +2,140 @@
...
+
+			// go down a row
+			sct += (n - 1);
 		}
+		sct = init + 1;
+		e++;
 	}
 
-	ofstream fout("fencedin.out") << ans << endl;
+	// naive kruskal's implementation
```
Comment (envyaims) [inline solutions/usaco-623.mdx:124]: wdym naive lmao

**Review by envyaims — CHANGES_REQUESTED** [review]

File: solutions/usaco-623.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/usaco-623.mdx:124]: oops

Comment (ryanchou-dev) [reply solutions/usaco-623.mdx]: [was typing my english essay while doing this (meant to type naive in the google doc)] 💀

**Review by envyaims — APPROVED** [review]
<3

**Review by brad182 — APPROVED** [review]

File: solutions/usaco-623.mdx [hunk 4]
```diff
@@ -2,115 +2,141 @@
...
+	/*
+	 * we'll number sections from left
+	 * to right.
+	 *
+	 * -------------
+	 * |  1  |  2  |
+	 * |-----------|
+	 * |  3  |  4  |
+	 * -------------
+	 */
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:93]:
```suggestion
	/*
	 * we'll number sections from left
	 * to right.
	 * 1 2
	 * 3 4
	 */
```

File: solutions/usaco-623.mdx [hunk 5]
```diff
@@ -2,115 +2,141 @@
...
-	ofstream fout("fencedin.out") << ans << endl;
+	// kruskal's
+	DSU dsu((n + 2) * (m + 2));
+
+	sort(edges.begin(), edges.end(), [](const Edge &t, const Edge &y) {
+		return t.cost < y.cost;
+	});
+
+	long long ans = 0;
+	for (auto &i : edges) {
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:133]: could you not use `auto`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
i think just thse two

**Comment by ryanchou-dev** [comment]
HI @brad182 OMGOMG

File: solutions/usaco-623.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/usaco-623.mdx:93]: what's wrong w this?

Comment (ryanchou-dev) [reply solutions/usaco-623.mdx:93]: feel like it makes more sense bc it shows the fences?

Comment (SansPapyrus683) [reply solutions/usaco-623.mdx:93]: takes too many lines imo

Comment (ryanchou-dev) [reply solutions/usaco-623.mdx:93]: eh, fair enough.

**Pull request merged**